packages/liquidator/src/liquidator.js

```diff
@@ -393,13 +346,29 @@ class Liquidator {
       message: "Checking for expired and disputed liquidations to withdraw rewards from"
     });
 
+    // In legacy versions of the EMP, withdrawing needs to be done by a party involved in the liquidation (i.e liquidator,
+    // sponsor or disputer).As the liquidator is the DSProxy, we would require the ability to send the withdrawal tx
+    // directly from the DSProxy to facilitate this.This functionality is not implemented as almost all legacy EMPs expired.
+    if (this.proxyTransactionWrapper.isUsingDsProxyToLiquidate && this.isLegacyEmpVersion) {
```

Comment:
@UMAprotocol/eng reviewers: not this decision here. If the bot is using DSProxy and the contract is legacy then withdrawing rewards is disabled. I chose to do this to keep the overall code churn and feature addition as simple as possible, especially seeing that almost all legacy contracts have expired. If a liquidation happens on a legacy contract we can still liquidate. However, we will need to manually withdraw from the DSProxy.

Reply:
@nicholaspai I think it's fine as there are basically no contracts running on the legacy version.

@mrice32 yup! we would need a very simple script to do this. I added a simple smart contract to this PR that we could use, if the need arises. it implements the correct interface for this contract version:

```
pragma solidity ^0.6.0;
pragma experimental ABIEncoderV2;

import "../../common/implementation/FixedPoint.sol";

// Simple contract used to withdraw liquidations using a DSProxy from legacy contracts (1.2.2 and below).
contract LiquidationWithdrawer {
    function withdrawLiquidation(
        address financialContractAddress,
        uint256 liquidationId,
        address sponsor
    ) public returns (IFinancialContract.RewardsData memory) {
        return IFinancialContract(financialContractAddress).withdrawLiquidation(liquidationId, sponsor);
    }
}

interface IFinancialContract {
    function withdrawLiquidation(uint256 liquidationId, address sponsor)
        external
        returns (FixedPoint.Unsigned memory amountWithdrawn);
}
```

Reply:
Is this tested?